crates/bevy_transform/src/hierarchy/hierarchy_maintenance_system.rs

```diff
@@ -45,31 +40,7 @@ pub fn parent_update_system(
         } else {
             commands.insert(entity, PreviousParent(parent.0));
         };
-
```

Comment:
Wont removing this break manual insertion of the `Parent(Entity)` component (or changing the parent value directly)? If so, that would be a pretty major breaking change.

Reply:
Yup theres no question that the whole system needs an overhaul. Currently it is possible (and easy) to mutably change Parent (either via commands or &mut Parent queries). Using that to change parents has probably been done (and regardless its _possible_ and easy to do). Breaking that behavior silently feels about as problematic as the bug being fixed in this pr. If we can do it in a way that completely prevents manual Parent mutations (both via queries and direct world access), thats worth considering.

I'm also starting to consider moving forward on merging my `hierarchy` branch, or if that ends up not being ready / too disruptive for 0.5, maybe just leaving this behavior as-is for 0.5.

 Let me know if you have any other suggestions.

Reply:
My preference would be to leave this behavior as-is and then do a full RFC on how parent-child stuff should work immediately after 0.5 lands. It's very central to UI, and @BoxyUwU's relations are also threatening to disrupt the space and need consideration.

Reply:
Cool, yeah there's several sets of changes in motion that will contribute to a better solution for this problem space. I don't have any good ideas about how to fix it in a way that works for everyone with what we have now, I think we need either Boxy's changes or something like #1655. I'm good closing this then, I'll keep running a fork until we get a proper solution.